### PR TITLE
Fix slide and slid event mixup

### LINKED ISSUE #73: Carousel Events are incorrect
The carousel emits two events: `slide` and `slid`.

The official bootstrap docs state that the `slide` event should fire first: http://getbootstrap.com/javascript/#carousel-events. (This makes sense, since `slid` is the past tense of `slide`.)

In bootstrap.native, the `slid` event fires first. This is incorrect.

Here are JSBins demonstrating the problem:

- jQuery bootstrap: http://jsbin.com/tapekeyata/edit?js,console,output
- bootstrap.native: http://jsbin.com/hutibawuyi/1/edit?js,console,output

Notice the order that the events fire. This is a bug.

@thednp Urgent :exclamation:

### PR DESCRIPTION
**Pull request by RyanZim**
Fixes #73.

My text editor trimmed trailing whitespace, sorry for the noisy diff.

Fixed the code and docs.

### CODE REVIEW

File: assets/js/scripts.js [hunk 1]
```diff
@@ -97,11 +97,11 @@ btn.addEventListener('click', function() {
...
 var genericCarousel = document.getElementById('carousel-example-generic');
-genericCarousel.addEventListener('slid.bs.carousel', function(e) {
-	console.log( 'The #' + e.target.id + ' is about to slide, and this was the "slid" event' );	
-});
 genericCarousel.addEventListener('slide.bs.carousel', function(e) {
-	console.log( 'The #' + e.target.id + ' has finished the slide transition, and this was the "slide" event' );	
+	console.log( 'The #' + e.target.id + ' is about to slide, and this was the "slide" event' );
+});
+genericCarousel.addEventListener('slid.bs.carousel', function(e) {
+	console.log( 'The #' + e.target.id + ' has finished the slide transition, and this was the "slid" event' );
```
Comment (RyanZim) [inline assets/js/scripts.js]: Changed the console message here.

File: dist/bootstrap-native.js [hunk 2]
```diff
@@ -402,7 +402,7 @@
           if ( self.options.interval !== false && !/\bpaused/.test(self.carousel.className) ){
             clearInterval(self.timer); self.cycle();
           }
-          if (slide) { self.carousel.dispatchEvent(slide); } //here we go with the slide
+          if (slid) { self.carousel.dispatchEvent(slid); } //here we go with the slid
```
Comment (RyanZim) [inline dist/bootstrap-native.js]: Changed the event here.

File: dist/bootstrap-native.js [hunk 3]
```diff
@@ -413,7 +413,7 @@
...
```
Comment (RyanZim) [inline dist/bootstrap-native.js]: And here.

File: dist/bootstrap-native.js [hunk 4]
```diff
@@ -370,13 +370,13 @@
...
       var slid = null, slide = null;
-      
+
       //register events
       if (('CustomEvent' in window) && window.dispatchEvent) {
         slid =  new CustomEvent("slid.bs.carousel");
         slide = new CustomEvent("slide.bs.carousel");
       }
-      if (slid) { this.carousel.dispatchEvent(slid); } //here we go with the slid
+      if (slide) { this.carousel.dispatchEvent(slide); } //here we go with the slide
```
Comment (RyanZim) [inline dist/bootstrap-native.js]: Changed event here

File: index.html [hunk 5]
```diff
@@ -910,27 +910,27 @@ <h4 class="panel-title" id="-collapsible-group-item-#3-">
...
+  var active = mainSlider.querySelector('.item.active .carousel-caption');
   active.classList.remove('slide')
 });
-mainSlider.addEventListener('slide.bs.carousel', function(e) {
+mainSlider.addEventListener('slid.bs.carousel', function(e) {
   // get the caption of new active item after slide
   var active = mainSlider.querySelector('.item.active .carousel-caption');
   active.classList.add('slide')
-});	
+});
```
Comment (RyanZim) [inline index.html]: Fixed docs here.

File: lib/carousel-native.js [hunk 6]
```diff
@@ -93,13 +93,13 @@
...
       var slid = null, slide = null;
-      
+
       //register events
       if (('CustomEvent' in window) && window.dispatchEvent) {
         slid =  new CustomEvent("slid.bs.carousel");
         slide = new CustomEvent("slide.bs.carousel");
       }
-      if (slid) { this.carousel.dispatchEvent(slid); } //here we go with the slid
+      if (slide) { this.carousel.dispatchEvent(slide); } //here we go with the slide
```
Comment (RyanZim) [inline lib/carousel-native.js]: Fixed event here

File: lib/carousel-native.js [hunk 7]
```diff
@@ -125,7 +125,7 @@
           if ( self.options.interval !== false && !/\bpaused/.test(self.carousel.className) ){
             clearInterval(self.timer); self.cycle();
           }
-          if (slide) { self.carousel.dispatchEvent(slide); } //here we go with the slide
+          if (slid) { self.carousel.dispatchEvent(slid); } //here we go with the slid
```
Comment (RyanZim) [inline lib/carousel-native.js]: and here

File: lib/carousel-native.js [hunk 8]
```diff
@@ -136,7 +136,7 @@
...
```
Comment (RyanZim) [inline lib/carousel-native.js]: and here

**Comment by thednp** [comment]
Ah so the Carousel.js script is fine then? Only the demo script.js is the issue? If so, please also change the `gh-pages` repository for the demo.

File: index.html (hunk 5 above)
Comment (RyanZim) [inline index.html]: Oops, wrong line, should have put this comment a few lines up

**Comment by RyanZim** [comment]
@thednp No, I just changed the console.log message there. I've commented on the other relevant changes.

**Comment by RyanZim** [comment]
Thanks! When are you planning on releasing this bugfix?

**Comment by thednp** [comment]
Well, I'm a little busy with other scripting, if you need this ASAP, please also update the dist folders and then also update the `gh-pages`.

**Comment by RyanZim** [comment]
@thednp dist folder updates were contained in this PR.

I'd be glad to update gh-pages,  how do you do it?

**Comment by thednp** [comment]
I do copy paste usually, I don't do extra geeky things you do 💃

**Comment by thednp** [comment]
Also please do an extensive test on your end before next PR.

**Comment by RyanZim** [comment]
Just tested it, it works great!

**Pull request merged**